partitioned-heat-conduction/solver-fenics/heat_pySDC_problem_class.py

```diff
@@ -0,0 +1,142 @@
+from fenics import TrialFunction, TestFunction, dx, assemble, inner, nabla_grad, DirichletBC, Constant, solve, interpolate
+
+from pySDC.core.Problem import ptype
+from pySDC.implementations.datatype_classes.fenics_mesh import fenics_mesh, rhs_fenics_mesh
+
+
+class fenics_heat_2d(ptype):
+    dtype_u = fenics_mesh
+    dtype_f = rhs_fenics_mesh
+
+    def __init__(self, mesh, function_space, forcing_term_expr, solution_expr, coupling_boundary,
+                 remaining_boundary, coupling_expr, precice_ref, participant_name):
+        # Add docstring
+        """
+        Constructor of the 2D heat equation problem class.
+
+        Args:
+            mesh: FEniCS mesh object
+            function_space: FEniCS function space object
+            forcing_term_expr: FEniCS expression for the forcing term
+            solution_expr: FEniCS expression for the manufactured solution
+            coupling_boundary: FEniCS SubDomain object for the coupling boundary
+            remaining_boundary: FEniCS SubDomain object for the remaining boundary
+            coupling_expr: FEniCS expression for the coupling boundary condition
+            precice_ref: preCICE-FEniCS adapter object reference
+            participant_name: Name of the participant (Dirichlet or Neumann)
+        """
+
+        # Set precice reference and coupling expression reference to update coupling boundary
+        # at every step within pySDC
+        self.precice = precice_ref
+        self.coupling_expression = coupling_expr
+        self.t_start = 0.0
+        self.participant_name = participant_name
```

Comment:
```suggestion
        self.participant_name = self.precice.get_participant_name()
```
Or completely remove `self.participant_name` and just call `self.precice.get_participant_name()`? The operation is not costly and it also helps you to avoid bugs if you avoid storing redundant data.